target_postgres/connector.py

```diff
@@ -255,12 +262,22 @@ def _handle_array_type(self, jsonschema: dict) -> ARRAY | JSONB:
         # Case 3: tuples
         return ARRAY(JSONB()) if isinstance(items, list) else JSONB()
 
+    def _handle_integer_type(self, jsonschema: dict) -> SMALLINT | INTEGER | BIGINT:
+        """Handle integer type."""
+        if maximum := jsonschema.get("maximum"):
+            if maximum < 2**15:
+                return SMALLINT()
+            if maximum < 2**31:
+                return INTEGER()
+
+        return BIGINT()
+
```

Comment:
I personally don't like inferring the desired integer type based on a "maximum" value.  
It's IMO unnecessarily indirect and also a bit imprecise, as the numeric Postgres integer types are signed, and the maximum would also indicate a "minimum", which however differs by one . For example, smallint is in the range`[-2**15, 2**15)`.

why not letting the schema specify it explicitly?

Something like:

```
{"type": integer, "dbtype": "smallint"}
```

(name of "dbtype" up for debate of course)

Reply:
I've pushed 5966233 to handle the ranges correctly.

Now, I _do_ want to support custom keywords like `dbtype` (see https://github.com/meltano/sdk/issues/2102).